src/server/cli.ts

```diff
@@ -0,0 +1,166 @@
+  .addOption(passwordOption)
+  .addOption(verboseOption)
+  .action((configPath, { username, password, serverRoot, serverHost, port, verbose }) => {
+    const anchorPath = path.dirname(configPath);
+    const auth = parseCredentials(username, password);
+    const config = loadConfigFile(configPath, program);
+    const { appSettings, sources, serverSettings } = buildSettings(config, { serverRoot, serverHost, verbose, port }, auth);
+    const settingsManager = new SettingsManager(appSettings, sources, {
+      anchorPath,
+      initialLoadTimeout: serverSettings.pageMustLoadTimeout,
+      verbose,
+      logger: LOGGER
+    });
+    createServer(serverSettings, createApp(serverSettings, settingsManager, version), program);
+  });
+
+program
+  .command("run-examples")
+  .addOption(portOption)
+  .addOption(serverRootOption)
+  .addOption(serverHostOption)
+  .addOption(verboseOption)
+  .action(({ port, verbose, serverRoot, serverHost }) => {
+    const configPath = path.join(__dirname, "../../config-examples.yaml");
+    const anchorPath = path.dirname(configPath);
+    const config = loadConfigFile(configPath, program);
+    const { sources, serverSettings, appSettings } = buildSettings(config, { port, verbose, serverHost, serverRoot });
+    const settingsManager = new SettingsManager(appSettings, sources, {
+      anchorPath,
+      initialLoadTimeout: serverSettings.pageMustLoadTimeout,
+      verbose,
+      logger: LOGGER
+    });
+    createServer(serverSettings, createApp(serverSettings, settingsManager, version), program);
+  });
+
+program
+  .command("connect-druid")
+  .argument("<url>", "druid url")
+  .addOption(portOption)
+  .addOption(serverRootOption)
+  .addOption(serverHostOption)
+  .addOption(verboseOption)
+  .addOption(usernameOption)
+  .addOption(passwordOption)
+  .action((url, { port, verbose, username, password, serverRoot, serverHost }) => {
+    const auth = parseCredentials(username, password);
+    const { appSettings, serverSettings, sources } = settingsForDruidConnection(url, { port, verbose, serverHost, serverRoot }, auth);
+    const settingsManager = new SettingsManager(appSettings, sources, {
+      anchorPath: process.cwd(),
+      initialLoadTimeout: serverSettings.pageMustLoadTimeout,
+      verbose,
+      logger: LOGGER
+    });
+    createServer(serverSettings, createApp(serverSettings, settingsManager, version), program);
+  });
+
+program
+  .command("load-file")
+  .argument("<file>", "json file")
+  .requiredOption("-t, --time-attribute <attribute>", "time attribute")
+  .addOption(portOption)
+  .addOption(serverRootOption)
+  .addOption(serverHostOption)
+  .addOption(verboseOption)
+  .action((file, { timeAttribute, port, verbose, serverHost, serverRoot }) => {
+    const { appSettings, sources, serverSettings } = settingsForDatasetFile(file, timeAttribute, { serverRoot, serverHost, verbose, port });
+    const settingsManager = new SettingsManager(appSettings, sources, {
```

Comment:
I would extract method for creating SettingsManager and Server and reuse for run-examples, connect-druid and load-file